### PR TITLE
Clean up '#if CORE' - use SecurityZone from CoreFX

### PR DESCRIPTION
**Pull request by iSazonov**
Related #4357 and #3565.

Remove the our stub for SecurityZone and use CoreFX implementation.

### CODE REVIEW

**Comment by iSazonov** [comment]
@SteveL-MSFT Invoke-Item.Tests.ps1 was failed on McOS 😕 Becouse it seems very-very slow.

File: src/System.Management.Automation/System.Management.Automation.csproj [hunk 1]
```diff
@@ -16,6 +16,7 @@
     <PackageReference Include="System.IO.FileSystem.AccessControl" Version="4.4.0-preview3-25423-02" />
     <PackageReference Include="System.Security.AccessControl" Version="4.4.0-preview3-25423-02" />
     <PackageReference Include="System.Security.Cryptography.Pkcs" Version="4.4.0-preview3-25423-02" />
+    <PackageReference Include="System.Security.Permissions" Version="4.4.0-preview3-25423-02" />
```
Comment (daxian-dbw) [inline src/System.Management.Automation/System.Management.Automation.csproj]: This is great! I will replace our `GetFileSecurityZone` implementation with `System.Security.Policy.CreateFromUrl(string url).SecurityZone`

File: src/System.Management.Automation/engine/Modules/ImportModuleCommand.cs [hunk 2]
```diff
@@ -14,6 +14,7 @@
 using System.Management.Automation.Internal;
 using System.Diagnostics.CodeAnalysis;
 using System.Security;
+using System.Security.Permissions;
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/Modules/ImportModuleCommand.cs]: `SecurityZone` is in the namespace `System.Security` ... so this one is not needed.

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs [hunk 3]
```diff
@@ -16,6 +16,7 @@
 using System.Management.Automation.Provider;
 using System.Security;
 using System.Security.AccessControl;
+using System.Security.Permissions;
```
Comment (daxian-dbw) [inline src/System.Management.Automation/namespaces/FileSystemProvider.cs]: Ditto.

File: src/System.Management.Automation/security/SecurityManager.cs [hunk 4]
```diff
@@ -12,13 +12,9 @@
 using System.Management.Automation.Host;
 using System.Management.Automation.Language;
 using System.Security;
+using System.Security.Permissions;
```
Comment (daxian-dbw) [inline src/System.Management.Automation/security/SecurityManager.cs]: Ditto.

File: src/System.Management.Automation/utils/ClrFacade.cs [hunk 5]
```diff
@@ -15,13 +15,13 @@
 using System.Text;
 using System.Threading;
 using System.Security;
+using System.Security.Permissions;
```
Comment (daxian-dbw) [inline src/System.Management.Automation/utils/ClrFacade.cs]: Ditto.

File: src/System.Management.Automation/engine/Modules/ImportModuleCommand.cs (hunk 2 above)
Comment (iSazonov) [inline src/System.Management.Automation/engine/Modules/ImportModuleCommand.cs]: Fixed.

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs (hunk 3 above)
Comment (iSazonov) [inline src/System.Management.Automation/namespaces/FileSystemProvider.cs]: Fixed.

File: src/System.Management.Automation/security/SecurityManager.cs (hunk 4 above)
Comment (iSazonov) [inline src/System.Management.Automation/security/SecurityManager.cs]: Fixed.

File: src/System.Management.Automation/utils/ClrFacade.cs (hunk 5 above)
Comment (iSazonov) [inline src/System.Management.Automation/utils/ClrFacade.cs]: Fixed.

**Review by daxian-dbw — APPROVED** [review]

File: src/System.Management.Automation/System.Management.Automation.csproj (hunk 1 above)
Comment (daxian-dbw) [inline src/System.Management.Automation/System.Management.Automation.csproj]: It turns out there is no corresponding implementation of `Zone.CreateFromUrl(string url)` ...
The implementation just returns null.

**Pull request merged**